text/1003-deprecation-import-ember-from-ember.md

```diff
@@ -0,0 +1,333 @@
+- `Ember.Instrumentation.subscribe`
+- `Ember.Instrumentation.unsubscribe`
+- `Ember.Instrumentation.reset`
+- `Ember.ViewUtils`
+- `Ember.ViewUtils.getChildViews`
+- `Ember.ViewUtils.getElementView`
+- `Ember.ViewUtils.getRootViews`
+- `Ember.ViewUtils.getViewBounds`
+- `Ember.ViewUtils.getViewBoundingClientRect`
+- `Ember.ViewUtils.getViewClientRects`
+- `Ember.ViewUtils.getViewElement`
+- `Ember.ViewUtils.isSimpleClick`
+- `Ember.ViewUtils.isSerializationFirstNode`
+
+
+Perhaps we can have folks add this to their apps:
+```js
+import { macroCondition, isDevelopingApp, importSync } from '@embroider/macros';
+
+if (macroCondition(isDevelopingApp())) {
+  // maybe this is side-effecting and installs 
+  // some functions on `globalThis` that the inspector could call
+  importSync('@ember/inspector-support');
+}
+```
+
+### No replacements.
+
+Applies to both the value and type exports (if applicable).
+
+- `Ember._getPath`
+- `Ember.isNamespace`
+- `Ember.toString`
+- `Ember.Container`
+- `Ember.Registry`
+
+Internal decorator utils
+- `Ember._descriptor`
+- `Ember._setClassicDecorator`
+
+Reactivity
+- `Ember.beginPropertyChanges`
+- `Ember.changeProperties`
+- `Ember.endPropertyChanges`
+
+Observable 
+- `Ember.hasListeners`
+
+Mixins
+- `Ember._ContainerProxyMixin`
+- `Ember._ProxyMixin`
+- `Ember._RegistryProxyMixin`
+- `Ember.ActionHandler`
+- `Ember.Comparable`
+
+Utility
+- `Ember.libraries` - 
+   App authors could choose to use any webpack or other build plugin that collections this information, such as [webpack-node-modules-list](https://github.com/ubilabs/webpack-node-modules-list). This additionally means that V1 libraries that pushed themselves into `Ember.libraries` no longer need to worry about interacting with this or any similar API. 
+- `Ember._Cache`
+- `Ember.GUID_KEY`
+- `Ember.canInvoke`  
+    Instead use [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining):
+    ```js
+    this.foo?.method?.();
+    ```
+- `Ember.testing`  
+  Instead, use
+
+  ```js
+  import { macroCondition, isTesting } from '@embroider/macros';
+
+  // ...
+
+  if (macroCondition(isTesting()) {
+    // test only code here
+  }
+  ```
+
+- `Ember.onerror`
+  Instead use an event listener for the `error` event on window.
+  ```js
+  window.addEventListener('error', /* ... event handler ... */);
+  ```
+- `Ember.generateGuid`
+- `Ember.uuid`
+- `Ember.wrap`
+- `Ember.FEATURES`  
+    But this is useful when working with feature flags.
+    This information could live on a specially named `globalThis` property, the enabled features could be emitted as a virtual module to import.
+- `Ember.ControllerMixin`
+- `Ember.deprecateFunc`
+- `Ember.inspect`
+- `Ember.Debug`
+  Replaced by some of `@ember/debug` exports.
+- `Ember.cacheFor`
+- `Ember.ComputedProperty`
+- `Ember.RouterDSL`
+- `Ember.controllerFor`
+- `Ember.generateController`
+- `Ember.generateControllerFactory`
+- `Ember.VERSION`  
```

Comment:
It's also required for inspector. It's in an amd module anyway.

Reply:
related: https://github.com/emberjs/rfcs/pull/1003#discussion_r1483686852